### ISSUE TITLE
Rework Device Configuration

### ISSUE THREAD

**Issue by danascape** [sworkflow-project/sworkflow#2]
* The tool uses local stored configs (https://github.com/sworkflow-project/sworkflow/tree/main/configs) to start builds for devices, reworking on a way to easily generate a config locally and use that for a kernel source

**Comment by ikcod** [comment]
I would like to take up this issue and work on streamlining the process for generating and using local configurations for sworkflow. Here's my understanding of the problem and a proposed solution:
### Understanding of the Issue:
The tool currently uses locally stored configuration files to start builds for devices, but there's no straightforward way to generate and apply a configuration tailored to a specific kernel source. This creates a gap in usability, especially for developers who may need a streamlined process to modify and use configs efficiently.
### Proposed Solution:
- Automated Config Generator: I plan to create a script or utility to dynamically generate configuration files based on user inputs like:
  - Device name.
  - Kernel source path.
  - Build-specific options such as compiler preferences (e.g., Clang, GCC).
- Integration with Existing Workflow:The tool's build commands can be enhanced to accept a path to the generated configuration, ensuring seamless integration without disrupting the current workflow.
- Validation Mechanism:A validation step will ensure that the generated config adheres to the expected format and contains all required parameters.

**Comment by danascape** [comment]
yes, that would be great, I am using a bizarre solution at the moment, with pre-defined config, you can work on it

**Comment by ikcod** [comment]
Got it, thanks for letting me know! I'll start working on a better solution to replace the current setup. Could you share a bit more about the existing configuration or any specific requirements you'd like me to consider?
